File: tutorial/worked-examples/texting-bans-replication-python/analysis/code/02_figures.py
# 02_figures.py — Event study plot
# =============================================================================
# Loads coefficients from CSV, adds reference period, and creates
# a publication-quality event study plot.
# =============================================================================

import pandas as pd
import matplotlib
matplotlib.use("Agg")
import matplotlib.pyplot as plt

# ── Load coefficients ────────────────────────────────────────
coef_df = pd.read_csv(ANALYSIS / "output" / "event_study_coefs.csv")

# Add reference period (t = -1, coefficient = 0)
ref_row = pd.DataFrame([{
    "event_time": -1,
    "coefficient": 0.0,
    "std_error": 0.0,
    "ci_lower": 0.0,
    "ci_upper": 0.0,
}])
coef_df = pd.concat([coef_df, ref_row], ignore_index=True).sort_values("event_time")

# ── Create plot ──────────────────────────────────────────────
fig, ax = plt.subplots(figsize=(10, 6))

# Reference lines
ax.axhline(0, color="red", linestyle="--", linewidth=0.8, alpha=0.7)
ax.axvline(-0.5, color="gray", linestyle="--", linewidth=0.8, alpha=0.5)

# Point estimates with error bars
ax.errorbar(
    coef_df["event_time"],
    coef_df["coefficient"],
    yerr=[
        coef_df["coefficient"] - coef_df["ci_lower"],
        coef_df["ci_upper"] - coef_df["coefficient"],
    ],
    fmt="D",
    color="navy",
    capsize=4,
    markersize=5,
    linewidth=1.5,
    capthick=1.2,
)

ax.set_xlabel("Years Relative to Texting Ban", fontsize=12)
ax.set_ylabel("Effect on Log Fatalities", fontsize=12)
ax.set_title("Event Study: Texting Bans and Traffic Fatalities", fontsize=14)

ax.set_xticks(range(-6, 7))
ax.tick_params(labelsize=10)

# Caption
ax.annotate(
    "Coefficients relative to t = \u22121. 95% CIs shown. Clustered SEs at state level.",
    xy=(0.5, -0.12), xycoords="axes fraction",
    ha="center", fontsize=9, color="gray",
)

plt.tight_layout()
out_path = ANALYSIS / "output" / "event_study.png"
plt.savefig(out_path, dpi=300, bbox_inches="tight")
plt.close()
print(f"    Saved plot to {out_path}")

File: tutorial/worked-examples/texting-bans-replication-r/analysis/code/02_figures.R
# 02_figures.R -- Event study plot
# =============================================================================

suppressPackageStartupMessages({
  library(tidyverse)
})

# ── Load coefficients ────────────────────────────────────────
coef_df <- read_csv(file.path(ANALYSIS, "output", "event_study_coefs.csv"),
                    show_col_types = FALSE)

# Add reference period (t = -1)
coef_df <- coef_df %>%
  bind_rows(tibble(event_time = -1, coefficient = 0,
                   std_error = 0, ci_lower = 0, ci_upper = 0)) %>%
  arrange(event_time)

# ── Create plot ──────────────────────────────────────────────
p <- ggplot(coef_df, aes(x = event_time, y = coefficient)) +
  geom_hline(yintercept = 0, linetype = "dashed", color = "red", alpha = 0.7) +
  geom_vline(xintercept = -0.5, linetype = "dashed", color = "gray", alpha = 0.5) +
  geom_pointrange(aes(ymin = ci_lower, ymax = ci_upper),
                  color = "navy", size = 0.5, shape = 18) +
  scale_x_continuous(breaks = -6:6) +
  labs(
    x = "Years Relative to Texting Ban",
    y = "Effect on Log Fatalities",
    title = "Event Study: Texting Bans and Traffic Fatalities",
    caption = "Coefficients relative to t = \u22121. 95% CIs shown. Clustered SEs at state level."
  ) +
  theme_minimal(base_size = 12) +
  theme(
    panel.grid.minor = element_blank(),
    plot.title = element_text(hjust = 0.5, size = 14),
    plot.caption = element_text(color = "gray50", size = 9)
  )

out_path <- file.path(ANALYSIS, "output", "event_study.png")
ggsave(out_path, p, width = 10, height = 6, dpi = 300)
cat(sprintf("    Saved plot to %s\n", out_path))
